### PR TITLE
Suggest using docker when testing against postgres

### PR DESCRIPTION
**Pull request by DMRobertson**
Quick drive-by.

### CODE REVIEW

File: docs/development/contributing_guide.md [hunk 1]
```diff
@@ -206,7 +206,24 @@ This means that we need to run our unit tests against PostgreSQL too. Our CI doe
...
+terminal:
+
+```shell
+docker run --rm -e POSTGRES_PASSWORD=mysecretpassword -e POSTGRES_USER=postgres -e POSTGRES_DB=postgress -p 5432:5432 postgres:14
+```
+
+Then in a second terminal, invoke `trial`:
+
+```shell
+`SYNAPSE_POSTGRES=1 SYNAPSE_POSTGRES_HOST=127.0.0.1 SYNAPSE_POSTGRES_USER=postgres SYNAPSE_POSTGRES_PASSWORD=mysecretpassword poetry run trial tests
```
Comment (squahtx) [inline docs/development/contributing_guide.md:221]: stray backtick?
```suggestion
SYNAPSE_POSTGRES=1 SYNAPSE_POSTGRES_HOST=127.0.0.1 SYNAPSE_POSTGRES_USER=postgres SYNAPSE_POSTGRES_PASSWORD=mysecretpassword poetry run trial tests
```

**Review by squahtx — APPROVED** [review]
LGTM!

Docker might complain if postgres is already running locally. Might be worth making a note of it?
The changes look good as-is anyway.

File: docs/development/contributing_guide.md (hunk 1 above)
Comment (DMRobertson) [reply docs/development/contributing_guide.md:221]: Arg, thank you. (I copied Patrick's example from synapse-dev)

File: docs/development/contributing_guide.md [hunk 2]
```diff
@@ -206,7 +206,32 @@ This means that we need to run our unit tests against PostgreSQL too. Our CI doe
...
+
+If you see an error like
+
+```
+docker: Error response from daemon: driver failed programming external connectivity on endpoint nice_ride (b57bbe2e251b70015518d00c9981e8cb8346b5c785250341a6c53e3c899875f1): Error starting userland proxy: listen tcp4 0.0.0.0:5432: bind: address already in use.
+```
+
+then something is already bound to port 5432. You're probably already running postgres locally.
+
+Once you have a postgres server runing, invoke `trial` in a second terminal:
```
Comment (squahtx) [inline docs/development/contributing_guide.md:226]:
```suggestion
Once you have a postgres server running, invoke `trial` in a second terminal:
```

Comment (DMRobertson) [reply docs/development/contributing_guide.md:226]: Argh. Sorry, I don't mean to use you as my linter!

**Review by squahtx — APPROVED** [review]
still lgtm, save for the typo

**Pull request merged**